x-pack/solutions/security/packages/kbn-scout-security/src/playwright/fixtures/test/page_objects/alerts_table.ts

```diff
@@ -5,48 +5,30 @@
  * 2.0.
  */
 
-import { ScoutPage, Locator } from '@kbn/scout';
+import { ScoutPage, Locator, expect } from '@kbn/scout';
 
 const PAGE_URL = 'security/alerts';
-const EXPAND_EVENT = 'expand-event';
 
 export class AlertsTablePage {
-  public expandAlertBtn: Locator;
+  public alertRow: Locator;
+  public alertsTable: Locator;
 
   constructor(private readonly page: ScoutPage) {
-    this.expandAlertBtn = this.page.testSubj.locator(EXPAND_EVENT);
+    this.alertRow = this.page.locator('div.euiDataGridRow');
+    this.alertsTable = this.page.testSubj.locator('alertsTable');
   }
 
   async navigate() {
-    await this.page.gotoApp(PAGE_URL);
+    return this.page.gotoApp(PAGE_URL);
   }
 
-  async expandFirstAlertDetailsFlyout() {
-    const maxAttempts = 100;
-    let attempts = 0;
+  async expandAlertDetailsFlyout(ruleName: string) {
+    await this.page.waitForLoadingIndicatorHidden();
+    await this.alertsTable.waitFor({ state: 'visible' });
+    // Filter alert by unique rule name
+    const row = this.alertRow.filter({ hasText: ruleName });
+    await expect(row, `Alert with rule '${ruleName}' is not displayed`).toBeVisible();
```

Comment:
Very clear error message! Optional: I would possibly add `in the alerts table` at the end to make it super clear that is where we look for the rule name. The error would then be:

> Alert with rule '${ruleName}' is not displayed in the alters table.